mesh/mesh.hpp

```diff
@@ -1960,6 +1962,21 @@ class Mesh
                           Array<IntegrationPoint>& ips, bool warn = true,
                           InverseElementTransformation *inv_trans = NULL);
 
+   /** @brief Computes geometric parameters associated with a Jacobian matrix
+       in 2D/3D. These parameters are
+       (1) Area/Volume,
+       (2) Aspect-ratio (1 in 2D, and 2 non-dimensional and 2 dimensional
+                         parameters in 3D. Dimensional parameters are used
+                         for target construction in TMOP),
+       (3) skewness (1 in 2D and 3 in 3D), and finally
+       (4) orientation (1 in 2D and 3 in 3D).
+    */
+   void GetGeometricParametersFromJacobian(const DenseMatrix &J,
```

Comment:
I was just trying to figure out how to categorize this method for another PR (#3217) and I was wondering if this needs to be in `Mesh`.

I could have missed something but it looks like the only `Mesh` data being used in this method are the `Dim` and `spaceDim` but these could be inferred from the size of `J`.

This method could be implemented as a stand alone function alongside `CalcAdjugate`, `CalcOrtho`, etc. in the `DenseMatrix` files. It has similar requirements and similar outputs to these these functions.

Just a thought...

Reply:
Speaking of the size of `J`, this is not checked in this method. Couldn't the column and entry access lead to seg faults?

Reply:
ping: @kmittal2

Reply:
@mlstowell agreed.. we could move it outside the `Mesh` class.. I think it might have ended up there because of how I started writing it.. but as-is, it might fit better in one of the tmop files (I prefer to still keep it in a file where there is some relevant context).